### PR TITLE
[Juhaer Adittya] iP

### PR DESCRIPTION
**Pull request by pasha-292**

### CODE REVIEW

File: src/main/java/duke/register/Parser.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    private String command;
+    private String[] commandSeparate;
+    private String taskType;
+    private String taskName;
+    private String date = "";
+    private String index;
+
+    public Parser(String command) {
+        this.command = command;
+        commandSeparate = command.split(" ");
```
Comment (clarlzx) [inline src/main/java/duke/register/Parser.java:21]: Maybe commandsSplittedByWhiteSpace or just commandsSplitted might be a clearer name?

File: src/main/java/duke/register/Storage.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+     * @return
+     * @throws DukeException
+     */
+    public TaskList load() throws DukeException {
+        try {
+            TaskList taskList = new TaskList();
+            Scanner fio = new Scanner(file);
+            while (fio.hasNextLine()) {
+                String line = fio.nextLine();
+                String[] command = line.split(" ");
```
Comment (clarlzx) [inline src/main/java/duke/register/Storage.java:53]: Maybe the plural variable name commands will be better for an array?

File: src/main/java/duke/task/DeadlineTask.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+ * This class handles the deadlineTasks
+ */
+public class DeadlineTask extends Task {
+    private LocalDate date;
+
+    public DeadlineTask(String task) {
+        super(task);
+    }
+
+    String[] divideCommand = task.split(" ");
```
Comment (clarlzx) [inline src/main/java/duke/task/DeadlineTask.java]: Perhaps a clearer name would be better, also maybe this string array could be placed in toString() since it is not needed anywhere else?

File: src/main/java/duke/task/EventTask.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+        for (int i = 4; i < divideCommand.length; i++) {
+            if (i == divideCommand.length - 1) {
+                time += divideCommand[i];
+            } else {
+                time += divideCommand[i] + " ";
+            }
+        }
+        date = LocalDate.parse(time);
+        String dateFormat = date.format(DateTimeFormatter.ofPattern("MMM dd yyyy"));
+        String taskRepresent = divideCommand[1] + " " + divideCommand[2]
```
Comment (clarlzx) [inline src/main/java/duke/task/EventTask.java]: Maybe taskDisplayStr or taskDetailsStr might be clearer in showing this string's functionality?

File: src/test/java/duke/task/DeadlineTaskTest.java [hunk 5]
```diff
@@ -0,0 +1,22 @@
+package duke.task;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+
+public class DeadlineTaskTest {
+    @Test
+    public void Test1() {
```
Comment (clarlzx) [inline src/test/java/duke/task/DeadlineTaskTest.java:10]: Perhaps naming the methods with this format:

whatIsBeingTested_descriptionOfTestInputs_expectedOutcome 
e.g., intDivision_zeroDivisor_exceptionThrown

would better show what is being tested.

File: src/main/java/duke/Duke.java [hunk 6]
```diff
@@ -0,0 +1,75 @@
+package duke;
```
Comment (nhzaci) [inline src/main/java/duke/Duke.java:1]: I like the use of packages for your project

File: src/main/java/duke/Duke.java [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+
+import duke.exception.DukeException;
+import duke.register.Storage;
+import duke.register.Ui;
+import duke.task.DeadlineTask;
+import duke.task.EventTask;
+import duke.task.TaskList;
+import duke.task.TodoTask;
+import duke.register.Parser;
```
Comment (nhzaci) [inline src/main/java/duke/Duke.java:4]: Perhaps you could consider leaving an empty space between imports from different packages?

File: src/main/java/duke/register/Parser.java [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+        taskName = commandSeparate[1] + " " + commandSeparate[2];
+    }
+
+    public void parseTaskDate() {
+        if (commandSeparate.length <= 3) {
+            date = "";
+        } else {
+            for (int i = 4; i < commandSeparate.length; i++) {
+                date += i == commandSeparate.length - 1
+                        ? commandSeparate[i] : commandSeparate[i] + " ";
```
Comment (nhzaci) [inline src/main/java/duke/register/Parser.java:38]: Perhaps you could follow the format below for a ternary operator:
```java
date += i == commandSeparate.length - 1
        ? commandSeparate[i]
        : commandSeparate[i] + " "
```

File: src/main/java/duke/register/Storage.java [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+     */
+    public Storage(String filePath) {
+        this.path = filePath;
+        file = new File(filePath);
+    }
+
+    /**
+     * Checks if the file exist
+     */
+    public void checkIfExist() {
```
Comment (nhzaci) [inline src/main/java/duke/register/Storage.java]: Perhaps you could consider naming it differently since checkIfExist() sounds like a boolean expression, however, it performs a stateful operation instead

File: src/main/java/duke/register/Storage.java [hunk 10]
```diff
@@ -0,0 +1,88 @@
...
+            file.createNewFile();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * Reads the data from the file and returns then in a task list
+     * @return
+     * @throws DukeException
```
Comment (nhzaci) [inline src/main/java/duke/register/Storage.java:45]: Perhaps you could consider adding a description on what gets returned and what triggers the Exception to be thrown?

File: src/main/java/duke/register/Ui.java [hunk 11]
```diff
@@ -0,0 +1,36 @@
+package duke.register;
+
+import java.util.Scanner;
+
+/**
+ * This class is mainly for the input
+ */
+public class Ui {
+    Scanner ui;
```
Comment (nhzaci) [inline src/main/java/duke/register/Ui.java:9]: Perhaps you could consider a different name for the Scanner variable?

File: src/main/java/duke/task/EventTask.java [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
+
+    public EventTask(String task) {
+        super(task);
+    }
+
+    String[] divideCommand = task.split(" ");
+
+    @Override
+    public String toString() {
+        String time = "";
```
Comment (nhzaci) [inline src/main/java/duke/task/EventTask.java]: Maybe you could consider a different variable name? Since the variable name of time could mislead another developer to think this is of the LocalDateTime type.

File: src/main/java/duke/task/Task.java [hunk 13]
```diff
@@ -0,0 +1,33 @@
+package duke.task;
+/**
+ * Basic task class from which the specific tasks are formed
+ */
+public class Task {
+    String task;
+    boolean done = false;
```
Comment (nhzaci) [inline src/main/java/duke/task/Task.java:12]: Perhaps you could consider adding the private access modifier to your variables?

File: src/main/java/duke/task/TaskList.java [hunk 14]
```diff
@@ -0,0 +1,96 @@
+package duke.task;
+
+import java.util.ArrayList;
+
+/**
+ * Class for storing the taskList which stores the tasks
+ */
+public class TaskList {
+    ArrayList<Task> Tasks;
```
Comment (nhzaci) [inline src/main/java/duke/task/TaskList.java:10]: Perhaps you could consider naming the variable name in camelCase instead of PascalCase?

File: src/main/java/duke/task/TaskList.java [hunk 15]
```diff
@@ -0,0 +1,96 @@
...
+    }
+
+    //Adds a task to the task list.
+    public void addTask(Task task) {
+        Tasks.add(task);
+        System.out.println("Got it! Added: \n" + task);
+        System.out.println("Now you have " + numOfTasks() + " items in your list");
+    }
+
+    //Second add method which does not print any message
```
Comment (nhzaci) [inline src/main/java/duke/task/TaskList.java]: Perhaps you could consider adding JavaDoc comments instead for public methods?

File: src/main/java/duke/task/Task.java [hunk 16]
```diff
@@ -0,0 +1,33 @@
...
+
+    Task(String task) {
+        this.task = task;
+    }
+
+    public void markDone() {
+        this.done = true;
+    }
+
+    public boolean getStatus() {
```
Comment (nhzaci) [inline src/main/java/duke/task/Task.java]: Perhaps you could consider naming this to sound more like a boolean?

File: text-ui-test/input.txt [hunk 17]
```diff
@@ -0,0 +1,13 @@
...
+deadline do homework /by Monday
+done 3
+event attend meeting /at 6:45pm
+todo make coffee
+todo clean room
+done 5
+list
+delete 5
+list
+bye
```
Comment (nhzaci) [inline text-ui-test/input.txt:13]: Perhaps you could consider adding test cases where you set the index to be out of range as well?? Maybe something like done 200, delete 500?

File: src/main/java/duke/task/DeadlineTask.java [hunk 18]
```diff
@@ -0,0 +1,37 @@
...
+            if (i == divideCommand.length - 1) {
+                time += divideCommand[i];
+            } else {
+                time += divideCommand[i] + " ";
+            }
+        }
+        date = LocalDate.parse(time);
+        String dateFormat = date.format(DateTimeFormatter.ofPattern("MMM dd yyyy"));
+        String taskRepresent = divideCommand[1] + " " + divideCommand[2]
+                + " (" + divideCommand[3].substring(1) + ": " + dateFormat + ")";
```
Comment (nhzaci) [inline src/main/java/duke/task/DeadlineTask.java]: I like that you broke the long String up into several lines

File: src/main/java/duke/task/TaskList.java [hunk 19]
```diff
@@ -0,0 +1,96 @@
...
+
+    //Returns a taskList of the tasks that have matching keyword.
+    public void findTask(String keyword) {
+        TaskList matches = new TaskList();
+        for (Task t : Tasks) {
+            String task = t.getTaskName();
+            String[] taskName = task.split(" ");
+            if (keyword.equals(taskName[0]) || keyword.equals(taskName[1])
+                    || keyword.equals(taskName[2])) {
+                matches.silentAdd(t);
```
Comment (nhzaci) [inline src/main/java/duke/task/TaskList.java]: I like the abstraction of silentAdd() to add your tasks instead of directly adding into the list.

**Review by nhzaci — COMMENTED** [review]
Looks good to merge! 👍 
Overall I found your code easy to read, except a few places where the variable names could be improved. Just a few nits to fix.
